### ISSUE TITLE
Expo52: Can not animate BlurView intensity with reanimated

### ISSUE THREAD

**Issue by ikryvorotenko** [expo/expo#32781]
### Minimal reproducible example

https://github.com/ikryvorotenko/expo52-blur

### What platform(s) does this occur on?

iOS

### Where did you reproduce the issue?

in a development build, in Expo Go

### Summary

BlurView wrapped into AnimatedComponent doesn't have an effect when passing sharedValue to intensity prop. 

Tried both passing `sharedValue` to intensity or using `useAnimatedProps`, no effect.

### Environment

```text
➜  expo52-blur git:(main) ✗ npx expo-env-info

  expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 15.0.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 18.18.0 - ~/.nvm/versions/node/v18.18.0/bin/node
      Yarn: 1.22.1 - ~/.nvm/versions/node/v18.18.0/bin/yarn
      npm: 9.8.1 - ~/.nvm/versions/node/v18.18.0/bin/npm
    Managers:
      CocoaPods: 1.15.2 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 24.1, iOS 18.1, macOS 15.1, tvOS 18.1, visionOS 2.1, watchOS 11.1
    IDEs:
      Xcode: 16.1/16B40 - /usr/bin/xcodebuild
    npmPackages:
      expo: ~52.0.2 => 52.0.2 
      expo-router: ~4.0.2 => 4.0.2 
      react: 18.3.1 => 18.3.1 
      react-dom: 18.3.1 => 18.3.1 
      react-native: 0.76.1 => 0.76.1 
      react-native-web: ~0.19.13 => 0.19.13 
    npmGlobalPackages:
      eas-cli: 12.6.2
    Expo Workflow: managed
```

### Expo Doctor Diagnostics

```text
➜  expo52-blur git:(main) ✗ npx expo-doctor@latest
Enabled experimental React Native Directory checks. Unset the EXPO_DOCTOR_ENABLE_DIRECTORY_CHECK environment variable to disable this check.
✔ Check package.json for common issues
✔ Check Expo config for common issues
✔ Check native tooling versions
✔ Check if the project meets version requirements for submission to app stores
✔ Check for common project setup issues
✔ Check dependencies for packages that should not be installed directly
✔ Check for app config fields that may not be synced in a non-CNG project
✔ Check for issues with Metro config
✔ Validate packages against React Native Directory package metadata
✔ Check npm/ yarn versions
✔ Check Expo config (app.json/ app.config.js) schema
✔ Check that native modules do not use incompatible support packages
✔ Check for legacy global CLI installed locally
✔ Check that packages match versions required by installed Expo SDK
✔ Check that native modules use compatible support package versions for installed Expo SDK

Didn't find any issues with the project!
```

**Comment by junchenjun** [comment]
Having the same issue

**Comment by ikryvorotenko** [comment]
hey @tsapeta, do you have any updates on this one?

**Comment by g5works** [comment]
I also have this problem, and can no longer animate the intensity of my blurview. I'm trying to animate it based on the scroll offset, and although this works with everything else which animates according to the scroll, blurview no longer works. The weird part is that it worked perfectly in Expo 51.

**Comment by DawidSmuga** [comment]
Same problem, waiting for the fix too :(

**Comment by tsapeta** [comment]
Hey, I'm investigating it right now. It seems that it happens only with the New Architecture enabled, can someone confirm?

**Comment by tsapeta** [comment]
I figured out this was an issue in `react-native-reanimated`. Luckily, something similar was reported a while ago and the PR fixing both issues was already open (and merged just now). I'll let you know here once this patch is out.

**Comment by ikryvorotenko** [comment]
> Hey, I'm investigating it right now. It seems that it happens only with the New Architecture enabled, can someone confirm?

yes, can confirm that. Thanks for the update!

**Comment by brandtnewlabs** [comment]
> Hey, I'm investigating it right now. It seems that it happens only with the New Architecture enabled, can someone confirm?

Having the issue without the new architecture being enabled.

**Comment by LouisCapelle** [comment]
yes same issue here, just updated with the `3.16.4` and issue still here

**Comment by tsapeta** [comment]
Yeah, unfortunately the fix didn't make it into `3.16.4`. I requested Reanimated team to release another patch version with that fix.

**Comment by LouisCapelle** [comment]
yes i see that after @tsapeta, but it still doesn't work in the latest release. I logged the entire animated props in a useAnimatedReaction and the intensity in the value object doesn't update..

This is my code : 

```
export const HomeHeader: FC<Props> = ({
  profile,
  streaks,
  animatedPosition,
}) => {
  const insets = useSafeAreaInsets()
  const colors = useColors()
  const intensity = useSharedValue(1)

  const animatedProps = useAnimatedProps(() => {
    return {
      intensity: intensity.value,
    }
  }, [animatedPosition])

  useAnimatedReaction(
    () => animatedPosition.value,
    (value) => {
      intensity.value = interpolate(value, [0, 30], [1, 80], 'clamp')
    },
    [animatedProps]
  )

  return (
    <AnimatedBlurView
      animatedProps={animatedProps}
      style={[
        styles.container,
        { paddingTop: insets.top, borderBottomColor: colors.veryLightGray },
      ]}
      tint="systemUltraThinMaterialDark"
    >
      <Profile {...{ profile }} />
      <StreakPill {...{ streaks }} />
    </AnimatedBlurView>
  )
}
```

I tried many things, but it seems really tricky

**Comment by tsapeta** [comment]
`3.16.5` with the fix is out!

@LouisCapelle Do you use Expo Go, development build or bare?

**Comment by LouisCapelle** [comment]
development build only @tsapeta, and yes i updated with the latest release 3.16.5

**Comment by LouisCapelle** [comment]
@tsapeta i think it's the problem is `useAnimatedProps`, it doesn't update depending on the updater which in my code is animated position. The value of intensity is still 1 at every re render. 

the result of the clg is : `{"initial": {"updater": {}, "value": {"intensity": 1}}, "viewDescriptors": {"add": [Function add], "remove": [Function remove], "shareableViewDescriptors": {"_animation": null, "_isReanimatedSharedValue": true, "addListener": [Object], "modify": [Object], "removeListener": [Object], "value": [Array]}}}`

```
export const HomeHeader: FC<Props> = ({
  profile,
  streaks,
  animatedPosition,
}) => {
  const insets = useSafeAreaInsets()
  const colors = useColors()

  const animatedProps = useAnimatedProps(() => {
    return {
      intensity: interpolate(animatedPosition.value, [0, 30], [1, 80], 'clamp'),
    }
  })

  useAnimatedReaction(
    () => animatedProps,
    (value) => {
      console.log(value)
    },
    [intensity]
  )

  return (
    <Animated.View>
      <AnimatedBlurView
        animatedProps={animatedProps}
        style={[
          styles.container,
          { paddingTop: insets.top, borderBottomColor: colors.veryLightGray },
        ]}
        tint="systemUltraThinMaterialDark"
      >
        <Profile {...{ profile }} />
        <StreakPill {...{ streaks }} />
      </AnimatedBlurView>
    </Animated.View>
  )
} 
```

**Comment by LouisCapelle** [comment]
@tsapeta do you have an idea to fix this ?

**Comment by Hosam-hsm** [comment]
@LouisCapelle Were you able to solve this?

**Comment by LouisCapelle** [comment]
Nope... I explored the the code and can't find a way to fix this... @Hosam-hsm

**Comment by DawidSmuga** [comment]
Any ideas for fix that? Before update to 52 worked, when moved to 52 and new architecture not working even with latest (3.16.6) reanimated.

**Comment by adminmne0001** [comment]
> Any ideas for fix that? Before update to 52 worked, when moved to 52 and new architecture not working even with latest (3.16.6) reanimated.

Also seeing this (RN 0.76.5), commenting to follow along.

**Comment by yolpsoftware** [comment]
Same problem. Please fix this, it is our only roadblock to New Architecture.

**Comment by jakex7** [comment]
Hey, I have an update on the PR! It has been merged into the Reanimated repository and cherry-picked for version `3.17.0`.  

Just a heads-up - you'll need to whitelist the prop to animate it. You can do this by adding:  
`Animated.addWhitelistedNativeProps({ intensity: true });`  
before creating the animated component. This should make it work in the new version.

Big thanks to @piaskowyk and the Reanimated team for their efforts on this!

<details>
<summary>Test</summary>

```js
import { BlurView } from "expo-blur";
import React, { useEffect } from "react";
import { Image, StyleSheet, Text, View } from "react-native";
import Animated, {
  useAnimatedProps,
  useSharedValue,
  withRepeat,
  withTiming,
} from "react-native-reanimated";

Animated.addWhitelistedNativeProps({ intensity: true });
const AnimatedBlurView = Animated.createAnimatedComponent(BlurView);

export default function Test() {
  const animatedIntensity = useSharedValue<number | undefined>(0);

  useEffect(() => {
    animatedIntensity.value = withRepeat(
      withTiming(100, { duration: 2000 }),
      -1,
      true
      );
  }, []);

  return (
    <View style={styles.container}>
      <View style={styles.innerContainer}>
        <Image
          style={styles.image}
          source={{ uri: "https://source.unsplash.com/300x300" }}
        />
        <Text style={styles.blurredText}>This text is blurred</Text>

        <AnimatedBlurView
          style={styles.blurView}
          tint={"dark"}
          intensity={animatedIntensity}
        >
          <Text style={styles.nonBlurredText}>text</Text>
        </AnimatedBlurView>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    justifyContent: "center",
    padding: 6,
  },
  innerContainer: {
    alignItems: "center",
    justifyContent: "center",
  },
  image: {
    width: 250,
    height: 250,
  },
  blurredText: {
    position: "absolute",
    padding: 10,
    backgroundColor: "rgb(120,20,20)",
    color: "white",
    fontWeight: "bold",
    fontSize: 20,
    borderRadius: 5,
  },
  blurView: {
    ...StyleSheet.absoluteFillObject,
    alignItems: "center",
    paddingTop: 20,
  },
  nonBlurredText: {
    paddingHorizontal: 10,
    paddingVertical: 4,
    backgroundColor: "rgb(120,20,20)",
    color: "white",
    fontWeight: "bold",
    fontSize: 18,
  },
});


```

</details>

**Comment by LouisCapelle** [comment]
I confirm, thanks to @piaskowyk, it works perfectly on the last version ! 
Great Work ! <3

**Comment by zollipaul** [comment]
> Hey, I have an update on the PR! It has been merged into the Reanimated repository and cherry-picked for version `3.17.0`.
> 
> Just a heads-up - you'll need to whitelist the prop to animate it. You can do this by adding: `Animated.addWhitelistedNativeProps({ intensity: true });` before creating the animated component. This should make it work in the new version.
> 
> Big thanks to [@piaskowyk](https://github.com/piaskowyk) and the Reanimated team for their efforts on this!
> 
> Test

using the "entering" property in above in AnimatedBlurView, the app crashed and exited though without any error message.

**Comment by yolpsoftware** [comment]
I tried it out by enabling the New Architecture and pulling `"react-native-reanimted": "^3.17.0"` into our Expo SDK 52 app.

In our app, the animation is coupled to a ScrollView - when you scroll down, the upper part is blurred. This is implemented by changing a shared value in the scroll handler, and then passing the shared value into the `BlurView`'s `animatedProps`.

When I run this with the Old Architecture, then the blurring is updated in every frame, meaning I have about 60fps (or whatever) rendering updates to the blurring.

With the New Architecture unfortunately, the animation still does not work (simulator with a development build, and on a Testflight build on a real device). The `BlurView`'s blur value gets updated, but only after some time, or when the user "releases" the scrolling. As long as the scrolling happens, there isn't a frame-by-frame update of the blurring. So it definitely is an improvement (in 3.16.7, the blur value did not change at all), but the real problem is not fixed, since we do not have any frame-by-frame updates to the blur value.

**Comment by jakex7** [comment]
@yolpsoftware 

It sounds like the issue with updating the `sharedValue` rather than the `BlurView` itself, as regular animations updates BlurView every frame. I recommend creating an issue in the Reanimated repo with minimal reproduction steps, so that they can quickly investigate the problem 

https://github.com/software-mansion/react-native-reanimated/issues

**Comment by 1pharaxh** [comment]
**Hi I'm still have issues with Animating the intensity. I am whitelisting the prop before I declare the animated blur view but still no luck : (

**Comment by 1pharaxh** [comment]
Hi @jakex7 I tried your suggestion but it only works on the web, does not work on iOS I have replicated the issue here 

https://snack.expo.dev/@1pharaxh/blurview-broken?platform=web

Thanks for the help!
